File: modules/x-engine-jsi-viewer/android/jsi-viewer/src/main/java/com/zkty/jsi/JSI_viewer.java
package com.zkty.jsi;


import com.zkty.jsi.xengine_jsi_viewer;
import com.zkty.nativ.core.NativeContext;
import com.zkty.nativ.core.NativeModule;
import com.zkty.nativ.jsi.bridge.CompletionHandler;
import com.zkty.nativ.viewer.CallBack;
import com.zkty.nativ.viewer.Iviewer;
import com.zkty.nativ.viewer.Nativeviewer;

public class JSI_viewer extends xengine_jsi_viewer {
    Nativeviewer iViewer;
    @Override
    protected void afterAllJSIModuleInited() {
        NativeModule module = NativeContext.sharedInstance().getModuleByProtocol(Iviewer.class);
        if (module instanceof Nativeviewer)
            iViewer = (Nativeviewer) module;
    }


    @Override
    public void _openFileReader(_0_com_zkty_jsi_viewer_DTO dto, CompletionHandler<StatusDTO> handler) {
        iViewer.openFileReader(dto.fileUrl,dto.fileType, dto.title,new CallBack() {
            @Override
            public void success(String msg) {
                StatusDTO statusDTO = new StatusDTO();
                statusDTO.resultMsg = msg;
                handler.complete(statusDTO);
            }
        });
    }
}

File: modules/x-engine-jsi-viewer/iOS/Class/JSI_viewer.m
//
//  JSI_viewer.m
//  viewer
//
//  Created by zk on 2020/9/7.
//  Copyright © 2020 x-engine. All rights reserved.


#import "JSI_viewer.h"
#import "JSIContext.h"
#import "XENativeContext.h"
#import "iViewer.h"
@interface JSI_viewer()
@property(nonatomic,strong) id<iViewer> iviewer;

@end

@implementation JSI_viewer
JSI_MODULE(JSI_viewer)

- (void)afterAllJSIModuleInited {
    self.iviewer = XENP(iViewer);
}

- (void)_openFileReader:(_openFileReader_com_zkty_jsi_viewer_0_DTO *)dto complete:(void (^)(StatusDTO *, BOOL))completionHandler {
    [self.iviewer openFileWithfileUrl:dto.fileUrl fileType:dto.fileType title:dto.title];
}
@end
